### PR TITLE
bpo-36778: cp65001 encoding becomes an alias to utf_8

### PR DESCRIPTION
**Pull request by vstinner**
https://bugs.python.org/issue36778

### CODE REVIEW

**Comment by vstinner** [comment]
I reproduced https://github.com/python/cpython/pull/13110#issuecomment-491095964 benchmark:

```
Mean +- std dev: [ref] 156 ns +- 3 ns -> [remove] 105 ns +- 3 ns: 1.48x faster (-32%)
```

**Comment by vstinner** [comment]
@methane: Are you ok to simply remove cp65001.py?

**Comment by paulmon** [comment]
I can verify that this change fixes the issue with test_startup_imports I found on Windows IoT Core ARM32 as expected.

Also, all test_site and test_codec tests pass on Windows IoT Core with a default codepage of 65001

**Review by methane — APPROVED** [review]
> @methane: Are you ok to simply remove cp65001.py?

Yes.

File: Doc/library/codecs.rst [hunk 1]
```diff
@@ -1106,8 +1106,7 @@ particular, the following variants typically exist:
 +-----------------+--------------------------------+--------------------------------+
 | cp1258          | windows-1258                   | Vietnamese                     |
 +-----------------+--------------------------------+--------------------------------+
-| cp65001         |                                | Windows only: Windows UTF-8    |
-|                 |                                | (``CP_UTF8``)                  |
+| cp65001         |                                | Alias to ``utf_8`` encoding    |
```
Comment (serhiy-storchaka) [inline Doc/library/codecs.rst:1109]: Add the versionchanged directive.

I think it is better to remove this row add add cp65001 to the list of utf-8 aliases.

File: Lib/test/test_codecs.py [hunk 2]
```diff
@@ -875,95 +875,6 @@ def test_surrogatepass_handler(self):
             b"abc\xed\xa0z".decode(self.encoding, "surrogatepass")
 
 
-@unittest.skipUnless(sys.platform == 'win32',
-                     'cp65001 is a Windows-only codec')
-class CP65001Test(ReadTest, unittest.TestCase):
```
Comment (serhiy-storchaka) [inline Lib/test/test_codecs.py:880]: Does utf-8 pass these tests?

Comment (methane) [reply Lib/test/test_codecs.py:880]: I confirmed utf-8 passes this test.

Comment (vstinner) [reply Lib/test/test_codecs.py:880]: > Does utf-8 pass these tests?

Sorry, I forgot to specify before I merged my PR that yes: I tested on my Windows 10 and the test still passed. But CP65001Test is now redundant with UTF8Ttest.

> I confirmed utf-8 passes this test.

Thanks for checking :-)

File: Doc/library/codecs.rst (hunk 1 above)
Comment (vstinner) [reply Doc/library/codecs.rst:1109]: I created PR #13240 for your doc change proposal.

**Comment by taleinat** [comment]
Shouldn't this comment in `CodePageTest` in Lib/test/test_codecs.py also be removed? It no longer makes any sense.
```python
class CodePageTest(unittest.TestCase):
    # CP_UTF8 is already tested by CP65001Test
    CP_UTF8 = 65001
```

**Comment by vstinner** [comment]
> Shouldn't this comment in CodePageTest in Lib/test/test_codecs.py also be removed? It no longer makes any sense.

Right. I wrote PR #13807 to remove the comment.

**Pull request merged**